### PR TITLE
Parameter value generation - parity with alfred

### PR DESCRIPTION
**Pull request by sshvans**
This adds additonal parameter value generation for params which were missing in taskcat, for parity with alfred.
Following are the changes:
- Added missing parameter values check.
- Added ignorecase for regex matching
- Separated out the parameter value generation to its own method.

### CODE REVIEW

File: taskcat/stacker.py [hunk 1]
```diff
@@ -660,6 +675,231 @@ def generate_uuid(self, uuid_type):
...
+                if gets3replace.search(param_value):
+                    url = self.regxfind(geturl_re, param_value)
+                    param_value = self.get_s3contents(url)
+                    if self.verbose:
+                        print("{}Raw content of url {}".format(D, url))
+                    parmdict['ParameterValue'] = param_value
+
+                if getkeypair_re.search(param_value):
+                    keypair = self.regxfind(getkeypair_re, param_value)
+                    param_value = 'cikey'
```
Comment (santiagocardenas) [inline taskcat/stacker.py]: This must come from some config where we would rotate it... or at some point generate it and delete it after testing.

Comment (sshvans) [inline taskcat/stacker.py]: Sure. I'll add a config file called config.ini to save these default values, load it once at the start of taskcat and use it wherever needed.

File: taskcat/stacker.py [hunk 2]
```diff
@@ -660,6 +675,231 @@ def generate_uuid(self, uuid_type):
...
+                    parmdict['ParameterValue'] = param_value
+
+                if getlicensebucket_re.search(param_value):
+                    licensebucket = self.regxfind(getlicensebucket_re, param_value)
+                    param_value = 'quickstart-ci-license'
+                    if self.verbose:
+                        print("{}Generating default license bucket {}".format(D, licensebucket))
+                    parmdict['ParameterValue'] = param_value
+
+                if getmediabucket_re.search(param_value):
```
Comment (avattathil) [inline taskcat/stacker.py]: For feature parity this is fine.  As a future update, I suggest using os.envs to override 
So pre-execution
`export TASKCAT_MEDIA-BUCKET='quickstart-ci-media'` 
self.getmediabucket should pull from os value.
Another option:  add a constant into object  `self._mediabucket` with (get and set methods)

File: taskcat/stacker.py [hunk 3]
```diff
@@ -660,6 +675,231 @@ def generate_uuid(self, uuid_type):
...
+                    mediaBucket = self.regxfind(getmediabucket_re, param_value)
```
Comment (avattathil) [inline taskcat/stacker.py]: Should follow the same logic as `get_mediabucket`

File: taskcat/stacker.py (hunk 1 above)
Comment (avattathil) [inline taskcat/stacker.py]: I think get_keypaiir warrants an args param 
`taskcat -c quckstact/ci/config.yml` **-k tonynv**

We should set the keypair in the object (with get and set methods)
For feature parity current code is fine

**Comment by avattathil** [comment]
Per our discussion, merging this PR to gain parity - so we can support existing projects.

**Comment by santiagocardenas** [comment]
Are these changes compatible with Alfred's genpass? Alfred's genpass doesn't have the letter after the password length and it also filters through a standard password enforcement regex.

**Pull request merged**